hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationExecutor.java

```diff
@@ -95,7 +99,15 @@
      * @throws java.lang.NullPointerException if op is null.
      * @throws IllegalThreadStateException    if the operation is not allowed to be run on the calling thread.
      */
-    void runOperationOnCallingThread(Operation op);
+    void runOnCallingThread(Operation op);
+
+    /**
+     * Tries to run the operation on the calling thread if possible. Otherwise offload it to a different thread.
```

Comment:
so is it sometimes sync and sometimes async? or does it wait for the operation to finish even when it's offloaded to a different thread?